pkgs/stdenv/generic/setup.sh

```diff
@@ -827,7 +827,7 @@ unpackFile() {
 unpackPhase() {
     runHook preUnpack
 
-    if [ -z ${srcs+"${srcs[@]}"} ]; then
+    if [ -z ${srcs-"${srcs[@]}"} ]; then
```

Comment:
Intentional?

Reply:
`${srcs-"${srcs[@]}"}` expands to the first element of the `srcs` array; surely that’s not the intention.

```suggestion
    if [ ${#srcs[@]} = 0 ]; then
```